File: packages/miniapp/ios/MacaronMiniappView.swift
import ExpoModulesCore
import MacaronCapacitorBridge
import WebKit

/// MacaronMiniappView - Expo 模組的 View 層
/// 
/// 這是一個薄層封裝，直接使用 MacaronCapacitorBridge Framework 提供的能力
class MacaronMiniappView: ExpoView {
    
    // MARK: - Properties
    
    private var capacitorBridge: MacaronCapacitorBridge!
    private var webView: WKWebView!
    
    // Event dispatchers for React Native
    let onLoadEnd = EventDispatcher()
    let onError = EventDispatcher()
    let onMessage = EventDispatcher()
    
    // MARK: - Initialization
    
    required init(appContext: AppContext? = nil) {
        super.init(appContext: appContext)
        setupBridge()
    }
    
    private func setupBridge() {
        // 使用 MacaronCapacitorBridge Framework
        capacitorBridge = MacaronCapacitorBridge()
        webView = capacitorBridge.createWebView()
        
        // 設置導航代理以處理事件
        webView.navigationDelegate = self
        
        // 添加到視圖層級
        addSubview(webView)
        setupConstraints()
        
        print("✅ MacaronMiniappView: Bridge initialized")
    }
    
    private func setupConstraints() {
        // 關閉自動調整大小掩碼轉換，以使用 Auto Layout
        webView.translatesAutoresizingMaskIntoConstraints = false
        
        NSLayoutConstraint.activate([
            webView.topAnchor.constraint(equalTo: topAnchor),
            webView.leadingAnchor.constraint(equalTo: leadingAnchor),
            webView.trailingAnchor.constraint(equalTo: trailingAnchor),
            webView.bottomAnchor.constraint(equalTo: bottomAnchor)
        ])
    }
    
    // MARK: - Public Methods (供 React Native 調用)
    
    func loadURL(_ url: URL) {
        capacitorBridge.loadURL(url)
    }
    
    func loadHTML(_ html: String, baseURL: URL? = nil) {
        capacitorBridge.loadHTML(html, baseURL: baseURL)
    }
    
    func evaluateJavaScript(_ script: String, callback: @escaping (Any?, Error?) -> Void) {
        capacitorBridge.evaluateJavaScript(script, completion: callback)
    }
    
    // MARK: - Lifecycle Management
    
    override func willMove(toWindow newWindow: UIWindow?) {
        super.willMove(toWindow: newWindow)
        
        if newWindow != nil {
            capacitorBridge.viewWillAppear()
        } else {
            capacitorBridge.viewWillDisappear()
        }
    }
    
    override func didMoveToWindow() {
        super.didMoveToWindow()
        
        if window != nil {
            capacitorBridge.viewDidAppear()
        } else {
            capacitorBridge.viewDidDisappear()
        }
    }
    
    deinit {
        capacitorBridge.cleanup()
    }
}

// MARK: - WKNavigationDelegate

extension MacaronMiniappView: WKNavigationDelegate {
    
    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
        onLoadEnd([String: Any]())
    }
    
    func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {
        onError([
            "error": error.localizedDescription
        ])
    }
    
    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
        onError([
            "error": error.localizedDescription
        ])
    }
}

File: packages/miniapp/android/src/main/java/com/macaron/miniapp/MacaronMiniappView.kt
package com.macaron.miniapp

import android.content.Context
import android.webkit.WebView
import com.macaron.capacitor.MacaronCapacitorBridge
import expo.modules.kotlin.AppContext
import expo.modules.kotlin.views.ExpoView

/**
 * MacaronMiniappView - Expo 模組的 View 層
 * 
 * 這是一個薄層封裝，直接使用 MacaronCapacitorBridge Library 提供的能力
 */
class MacaronMiniappView(context: Context, appContext: AppContext) : ExpoView(context, appContext) {
    
    private val capacitorBridge: MacaronCapacitorBridge
    private val webView: WebView
    
    init {
        // 使用 MacaronCapacitorBridge Library
        capacitorBridge = MacaronCapacitorBridge(context)
        webView = capacitorBridge.createWebView()
        
        // 添加到視圖
        addView(webView)
        
        println("✅ MacaronMiniappView: Bridge initialized")
    }
    
    // MARK: - Public Methods (供 React Native 調用)
    
    fun loadURL(url: String) {
        capacitorBridge.loadURL(url)
    }
    
    fun loadHTML(html: String, baseURL: String = "capacitor://localhost") {
        capacitorBridge.loadHTML(html, baseURL)
    }
    
    fun evaluateJavaScript(script: String, callback: ((String?) -> Unit)? = null) {
        capacitorBridge.evaluateJavaScript(script) { result ->
            callback?.invoke(result)
        }
    }
    
    // MARK: - Lifecycle Management
    
    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        capacitorBridge.onStart()
        capacitorBridge.onResume()
    }
    
    override fun onDetachedFromWindow() {
        capacitorBridge.onPause()
        capacitorBridge.onStop()
        super.onDetachedFromWindow()
    }
    
    /**
     * 清理資源
     */
    fun cleanup() {
        capacitorBridge.cleanup()
    }
}
